algorithms/kernel/dtrees/dtrees_model_impl.h

```diff
@@ -466,6 +491,7 @@ class ModelImpl
         {
             arch->setSharedPtrObj(_impurityTables);
             arch->setSharedPtrObj(_nNodeSampleTables);
+            arch->setSharedPtrObj(_probTbl);
```

Comment:
move it under separate if branch
daalVersion >= COMPUTE_DAAL_VERSION(2020, 0, 0)